sdk/core/Azure.Core/src/Internal/SequenceWriterAccess.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.IO;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Serialization;
+
+namespace Azure.Core.Internal
+{
+    internal sealed class SequenceWriterAccess
```

Comment:
we should add a comment describing why we have this type and which stress tests need to be run when this file changes.

Reply:
Should we call this type ModelWriter and make SequenceWriter private nested type. This way nobody could actually use the writer without going through the RWLock.